go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1280,6 +1281,10 @@ func initSvcViaMgmPortRoutingRules(hostSubnets []*net.IPNet) error {
 		}
 	}
 
+	// lastly update the sysctl reverse path filtering options for ovn-k8s-mp0 interface to avoid dropping return packets
+	_, _ = sysctl.Sysctl(fmt.Sprintf("net/ipv6/conf/%s/rp_filter", types.K8sMgmtIntfName), "2")
```

Comment:
Does rp_filter for IPv6 even exist?

I don't have it on my systems:

```
$ sudo sysctl net.ipv6.conf.all.rp_filter
sysctl: cannot stat /proc/sys/net/ipv6/conf/all/rp_filter: No such file or directory
$ sudo sysctl net.ipv6.enp0s31f6.all.rp_filter
sysctl: cannot stat /proc/sys/net/ipv6/enp0s31f6/all/rp_filter: No such file or directory
$ 
```

Reply:
oh good point... I just went with the let's do it for ipv6 as well logic. damn so wondering how does this behave on ipv6 env? does it pick the settings from ipv4?

Reply:
hmm seems like I need to do something different for ipv6: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-securing_network_access#sect-Security_Guide-Server_Security-Reverse_Path_Forwarding

Reply:
All this rpf should be in the firewall ;-)  It was a mistake to have this sysctl for IPv4.  The sysctl used to be binary... but quickly broke things when a node became dual homed.  Since IP addresses belong to the node, and not any specific interface, the path you send to $dstIP typically isn't the path $dstIP uses sends to you.

Reply:
thanks Mike! this is a good learning lesson, let me do the right way of fixing this up!

Reply:
```
[496996.707078] x_tables: ip_tables: rpfilter match: used from hooks OUTPUT, but only valid from PREROUTING
```
running into this, gah investigating how to enable rp filter for locally generated traffic.

Reply:
```
ip6tables -t raw -N RPFILTER
ip6tables -t raw -A RPFILTER -m rpfilter -j RETURN
ip6tables -t raw -A RPFILTER -m rpfilter --loose -j ACCEPT
ip6tables -t raw -A PREROUTING -j RPFILTER 
```
hasn't worked. still looking.

Reply:
Guess what?! I just tested this on OCP IPV6 env and it works out of the box! :). Seems like the security is only being done for IPV4.
```
[surya@hidden-temple yaml_debugging]$ oc get pods -owide 
NAME                             READY   STATUS    RESTARTS   AGE   IP               NODE                                                NOMINATED NODE   READINESS GATES
hello-world-2-7f4f9dd6b9-lqdvn   1/1     Running   0          17s   fd01:0:0:5::12   worker-00.qe-anurag7c.qe.devcluster.openshift.com   <none>           <none>
hello-world-2-7f4f9dd6b9-qfl78   1/1     Running   0          17s   fd01:0:0:4::1a   worker-01.qe-anurag7c.qe.devcluster.openshift.com   <none>           <none>
```
```
[surya@hidden-temple yaml_debugging]$ oc get svc
NAME            TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
hello-world-2   ClusterIP   fd02::91b9   <none>        80/TCP    21s
```
```
[surya@hidden-temple yaml_debugging]$ oc debug node/worker-00.qe-anurag7c.qe.devcluster.openshift.com
Starting pod/worker-00qe-anurag7cqedevclusteropenshiftcom-debug ...
To use host binaries, run `chroot /host`
Pod IP: 2604:1380:4642:7e00::2d
If you don't see a command prompt, try pressing enter.
sh-4.4# curl [fd02::91b9]:80
Hello Kubernetes!sh-4.4# 
```
wooohoo, so need not touch any for IPV6 I guess :D

Reply:
setup on v6 looks good:
```
sh-4.4# ip -6 rule
0:      from all lookup local
30:     from all fwmark 0x1745ec lookup 7
32766:  from all lookup main
sh-4.4# ip -6 r show table 7
fd02::/112 via fd01:0:0:5::1 dev ovn-k8s-mp0 metric 1024 pref medium
sh-4.4#  ip6tables-save -c                                                                                                                                                   
[6:516] -A OVN-KUBE-ITP -d fd02::91b9/128 -p tcp -m tcp --dport 80 -j MARK --set-xmark 0x1745ec/0xffffffff
```